client/python/examples/general.py

```diff
@@ -0,0 +1,173 @@
+from armada_client.k8s.io.apimachinery.pkg.api.resource import (
+    generated_pb2 as api_resource,
+)
+from armada_client.typings import EventType
+
+
+def create_dummy_job(client: ArmadaClient):
+    """
+    Create a dummy job with a single container.
+    """
+
+    # For infomation on where this comes from,
+    # see https://github.com/kubernetes/api/blob/master/core/v1/generated.proto
+    pod = core_v1.PodSpec(
+        containers=[
+            core_v1.Container(
+                name="container1",
+                image="index.docker.io/library/ubuntu:latest",
+                args=["sleep", "10s"],
+                securityContext=core_v1.SecurityContext(runAsUser=1000),
+                resources=core_v1.ResourceRequirements(
+                    requests={
+                        "cpu": api_resource.Quantity(string="120m"),
+                        "memory": api_resource.Quantity(string="510Mi"),
+                    },
+                    limits={
+                        "cpu": api_resource.Quantity(string="120m"),
+                        "memory": api_resource.Quantity(string="510Mi"),
+                    },
+                ),
+            )
+        ],
+    )
+
+    return [client.create_job_request_item(priority=1, pod_spec=pod)]
+
+
+def wait_for_job_event(client, event_stream, job_id: str, event_state: EventType):
+    """
+    Wait for a job event to occur.
+
+    Will automatically return if the job fails or the event stream is closed.
+    """
+
+    # Contains all the possible message types
+    for event in event_stream:
+
+        event = client.unmarshal_event_response(event)
+
+        if event.message.job_id == job_id:
+            if event.type == EventType.failed:
+                return False
+            elif event.type == event_state:
+                return True
+
+
+def creating_jobs_example(client, queue, job_set_id):
+    """
+    Creates a jobset and job, and makes sure the job completes successfully.
+    """
+
+    # Create the PodSpec for the job
+    job_request_items = create_dummy_job(client)
+
+    resp = client.submit_jobs(
+        queue=queue, job_set_id=job_set_id, job_request_items=job_request_items
+    )
+
+    # Gets the job_id of the first job in the job_request_items
+    job_id = resp.job_response_items[0].job_id
+
+    client.reprioritize_jobs(new_priority=2, queue=queue, job_set_id=job_set_id)
+
+    # Needed to allow for the delay in the job_set being created
+    time.sleep(2)
+
+    # Event stream for the job_set
+    # Can be accessed directly as an iterator that will yield the next event
+    event_stream = client.get_job_events_stream(queue=queue, job_set_id=job_set_id)
+
+    test = wait_for_job_event(client, event_stream, job_id, EventType.succeeded)
+    if test:
+        print("Job submitted")
+
+    elif not test:
+        print("Failed")
+
+    # Close the event stream
+    client.unwatch_events(event_stream)
+
+
+def creating_queues_example(client, queue):
```

Comment:
Should we also do a more complicated queue with all the parameters?  It's be nice to explain the other queues and what they mean in code.

Reply:
I think doing this in a seperate file definitely makes sense. I think I would need a hand understanding the service and ingress configs though.

Reply:
Same!